tests/integration/test_usage_tracking_integration.py

```
@@ -0,0 +1,119 @@
+# Your actual, implemented components
+from tunacode.core.agents.main import _process_node
+from tunacode.core.state import StateManager
+from tunacode.core.token_usage.api_response_parser import ApiResponseParser
+from tunacode.core.token_usage.cost_calculator import CostCalculator
+from tunacode.configuration.models import ModelRegistry
+# Import the new UsageTracker class
+from tunacode.core.token_usage.usage_tracker import UsageTracker
+
+
+@pytest.mark.asyncio
+async def test_node_processing_updates_usage_state():
```

Pylint (3.3.7)

[refactor] Too many local variables (17/15)

(R0914)